pdns/backends/gsql/gsqlbackend.cc

```diff
@@ -1013,7 +1041,11 @@ bool GSQLBackend::get(DNSResourceRecord &r)
     } catch (SSqlException &e) {
       throw PDNSException("GSQLBackend get: "+e.txtReason());
     }
-    extractRecord(row, r);
+    try {
+      extractRecord(row, r);
+    } catch (...) {
+      goto skiprow;
```

Comment:
For records the problem is theoretical. It is unlikely a backend will return a broken DNSName if you ask for a good one in a query. The real problem lies in broken zones tsigkeys and metadata. 
One broken dnsname in domains for example is  enough to kill the masterslavecomminicator in the current master. 

This pull is not intended as the ultimate solution. It is just buying us some time to fix things for real while people enjoy a more or less working alpha release.